### PR TITLE
feat: improvements to the contributing guide

### PR DESCRIPTION
**Pull request by honzajavorek**
Reading through the contributing guide, I suggest the following improvements:

- Added top-level overview of the docs so that it's clear what "platform" or "academy" means when they're mentioned later
- Some paragraphs under felt like they belong under headings right after them
- Mention of redirects and SEO juice in the README felt oddly specific

I just fixed what I felt like fixing, didn't do a thorough proofreading or any effort to make this document beautiful, as I don't think that's the focus right now.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,35 +2,41 @@
 
 ## Architecture
 
-Apify documentation consists of six different repositories:
+Apify documentation consists of the following top-level areas:
 
+- **Academy** - Collection of mostly platform-independent courses on scraping techniques.
```
Comment (honzajavorek) [inline CONTRIBUTING.md:7]: Aspirational description 😅

File: README.md [hunk 2]
```diff
@@ -41,20 +41,19 @@ Instead, try to be more descriptive, for example:
...
-    const docsAreCool = require('coolDocs'); <br/>
-    ...<br/>
-    return docsAreCool;<br/>
+  const docsAreCool = require('coolDocs'); <br/>
+  ...<br/>
+  return docsAreCool;<br/>
 
-    \`\`\`
-    ```
+  \`\`\`
```
Comment (honzajavorek) [inline README.md:53]: This renders with the backslashes, but I didn't figure out how to do it in a correct way, so I left it as it was.

Comment (B4nan) [reply README.md:53]: https://stackoverflow.com/questions/42319711/fenced-code-block-inside-another-fenced-code-block

Comment (honzajavorek) [reply README.md:53]: I see I gave up too early! With all the AI around one would forget about the good old StackOverflow 😀 Thanks! Interesting solution, lol, adding more backticks. And it seems to work! At least in VS Code Markdown preview.

**Review by TC-MO — APPROVED** [review]

**Pull request merged**